### PR TITLE
Refactor transfer operator unit tests

### PR DESCRIPTION
**Pull request by pazner**
The unit tests in #2770 had relatively long runtime, this PR reduces the cases that are tested and slightly refactors the test code.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2829](https://github.com/mfem/mfem/pull/2829) | @pazner | @pazner | @psocratis + @pazner | 2/14/22 | 2/15/22 | ⌛due 3/7/22 | |

### CODE REVIEW

**Comment by pazner** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by psocratis — APPROVED** [review]
Looks great! Thanks @pazner

File: tests/unit/fem/test_transfer.cpp [hunk 1]
```diff
@@ -75,555 +75,482 @@ void vectorcoeff(const Vector& x, Vector& y)
...
+
+   int fineOrder = geometric ? order : 2*order;
+
+   INFO("  Vector space: " << VecSpaceName(vectorspace) << '\n' <<
+        "  Dimension:    " << dimension << '\n' <<
+        "  Simplex:      " << simplex << '\n' <<
+        "  Elements:     " << std::pow(ne, dimension) << '\n' <<
+        "  Coarse order: " << order << '\n' <<
+        "  Fine order:   " << fineOrder << '\n' <<
+        "  Geometric:    " << geometric);
```
Comment (YohannDudouit) [inline tests/unit/fem/test_transfer.cpp:111]: Nice! Do you know if there is many other places left with spurious `std::cout`?

**Review by YohannDudouit — APPROVED** [review]
Thanks @pazner ! Big improvement in code readability and removing the spurious `std::cout` is very much appreciated!

File: tests/unit/fem/test_transfer.cpp (hunk 1 above)
Comment (pazner) [reply tests/unit/fem/test_transfer.cpp:111]: Yeah, unfortunately I think there are quite a few other places that still remain.

**Review by drzisga — APPROVED** [review]

**Comment by pazner** [comment]
Merged in `next` for testing...

**Pull request merged**